packages/firebase_crashlytics/lib/src/firebase_crashlytics.dart

```diff
@@ -0,0 +1,190 @@
+// Copyright 2019, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+part of firebase_crashlytics;
+
+/// The entry point for accessing Crashlytics.
+///
+/// You can get an instance by calling `Crashlytics.instance`.
+class Crashlytics {
+  static final Crashlytics instance = Crashlytics();
+
+  /// Set to true to have errors sent to Crashlytics while in debug mode. By
+  /// default this is false.
+  bool enableInDevMode = false;
+
+  /// Keys to be included with report.
+  @visibleForTesting
+  final Map<String, dynamic> keys = <String, dynamic>{};
+
+  /// Logs to be included with report.
+  @visibleForTesting
+  final ListQueue<String> logs = ListQueue<String>(15);
+  int _logSize = 0;
+
+  bool get isInDebugMode {
+    bool _inDebugMode = false;
+    if (!enableInDevMode) {
+      assert(_inDebugMode = true);
+    }
+    return _inDebugMode;
+  }
+
+  static const MethodChannel channel =
+      MethodChannel('plugins.flutter.io/firebase_crashlytics');
+
+  Future<void> onError(FlutterErrorDetails details) async {
+    print('Error caught by Crashlytics plugin:');
+    if (isInDebugMode && !enableInDevMode) {
+      print(Trace.format(details.stack).trimRight().split('\n'));
+    } else {
+      // Send logs
+      await sendLogs();
+
+      // Send keys
+      await sendKeys();
+
+      // Report error
+      final List<String> stackTraceLines =
+          Trace.format(details.stack).trimRight().split('\n');
+      final dynamic result =
+          // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
+          // https://github.com/flutter/flutter/issues/26431
+          // ignore: strong_mode_implicit_dynamic_method
+          await channel.invokeMethod('Crashlytics#onError', <String, dynamic>{
+        'exception': details.exceptionAsString(),
+        'stackTrace': details.stack.toString(),
+        'stackTraceLines': stackTraceLines,
+        'code': stackTraceLines[0].hashCode
+      });
+      print(result);
+    }
+  }
+
+  void crash() {
+    throw StateError('Error thrown by Crashlytics plugin');
```

Comment:
Should this not delegate to the native SDK methods to ensure a full fatal crash (app terminates) as Crashlytics only submits reports after a termination and subsequent app startup - wouldn't the Dart/Flutter error handler capture this in debug/dev and thus preventing this termination behaviour? And dev/debug is where you'd want to be testing crashes.

**iOS**: `[[Crashlytics sharedInstance] crash];`
**Android**: `(new CrashTest()).crashAsyncTask(50);` (crash off main thread to avoid being captured by error handlers in dev/debug)

Reply:
@Salakar This was considered, however most Flutter level crashes don't result in the underlying native application crashing, so throwing an error here was an attempt to represent the types of crashes likely to happen an a Flutter app and how the report will ultimately be submitted to Crashlytics.